cats_frontend/src/app/modules/cats/services/cats.service.ts

```diff
@@ -11,76 +12,37 @@ export class CatsService {
     constructor(private http: HttpClient) { }
 
     get(
-        page?: number,
-        search?: string,
-        field?: string,
-        filter?: { [key: string]: {value: string, matchMode: string}}
+        params?: IMetaParams,
     ): Observable<{ cats: ICat[], count: number}> {
 
+        const apiParams: IApiParams = {};
 
-        const params: {[key: string]: any} = {};
-
-        if(page) {
-            params['page'] = page;
+        if(params?.page) {
+            apiParams['page'] = params.page.toString();
         }
 
-        if(search) {
-            params['search'] = search;
+        if(params?.search) {
+            apiParams['search'] = params.search;
         }
 
-        if(field) {
-            params['ordering'] = field;
+        if(params?.field) {
+            apiParams['ordering'] = params.field;
         }
 
-        if(filter) {
```

Comment:
you could simply do something like this :

 if (params?.filter) {
            const { filter } = params;
            for (const key in filter) {
                const { value, matchMode } = filter[key];
                if (value) {
                    apiParams[`${key}__${matchMode}`] = value;
                }
            }
        }